### ISSUE TITLE
[RPi4] Weird visual artefact in SCUMMVM games

### ISSUE THREAD

**Issue by beudbeud** [libretro/scummvm#73]
Hi, on some games like Zack mcKraken and India Jones and the fate of Atlantis i have some weird visual artifact 

![5904547247440380978](https://github.com/user-attachments/assets/6af63e87-3266-4fe9-a4c2-135b36a01ad9)
![5904547247440380977](https://github.com/user-attachments/assets/f0461f12-eece-4795-8341-c89faad48bcf)
![5904547247440380976](https://github.com/user-attachments/assets/0aa7498a-b6a2-4dfe-9f0d-a1b8222c2cdd)
![5904547247440380975](https://github.com/user-attachments/assets/74f707b9-7633-4613-92b0-d53768de9116)

**Comment by beudbeud** [comment]
I build core for raspberry pi 4 with the commit `e1a7e2931877937a04e55f45a9eec6e6f7d471ff` with `FORCE_OPENGLES2=1` options

**Comment by spleen1981** [comment]
Can't replicate this...
Can you provide a retroarch.log with debug level set both for core and frontend?
Was it working at any previous commit on that platform?

**Comment by beudbeud** [comment]
Before i use commit `65184d8aa28b403c492d9f2e338473a25c66a99b` and i have the same problem.

[retroarch.log](https://github.com/user-attachments/files/17845719/retroarch.log)

I have this kind of problem only SCUMM games, i try `Future wars` and `Discworld` i haven't this issue.

I will try one other version of Raspeberry pi

**Comment by beudbeud** [comment]
I use retroarch by KMS + OpenGLES

**Comment by spleen1981** [comment]
The log is debug level for core but info level for frontend. Can you set debug level for both?

> ScummVM core version: 9.2.2-pulsar-238-gd112796fdb

Not clear which version of the core are you testing, and if it is patched or not. I can see patches [here](https://gitlab.com/recalbox/recalbox/-/tree/master/package/libretro-scummvm?ref_type=heads), if you are using their build system. Including patches to scummvm engine. That would not be supported.

- So be sure the core is unpatched and @ latest commit. You can check from the log as above. You can also crossbuild outside their build system using your clone with:
```
cd backends/platform/libretro
make platform=rpi4_64 TOOLSET=<full_path_to_their_toolchain_bins>/<gcc_prefix(e.g. arm-linux-gnueabihf-)>
```

- Without this CRT mode the issue is still present?
- If you switch to sw rendering from core options (need to close the core and reopen to apply) the issue is still present?

**Comment by beudbeud** [comment]
[retroarch.log](https://github.com/user-attachments/files/17851306/retroarch.log)
[scummvm_libretro-2.txt](https://github.com/user-attachments/files/17851367/scummvm_libretro-2.txt) -> my retroarch configuration


I build  the core from last commit and the command you recommend me to use, and i got same result.

I can't try on HDMI, but i asked at other people and on rpi5 on HMDI my friend got same problem

Even i try to change the video_driver to sdl2 it's look not take effect and i have same problem.

**Comment by spleen1981** [comment]
Can you try disabling HW acceleration from the core video options?

**Comment by beudbeud** [comment]
i had `scummvm_video_hw_acceleration = "disabled"` and still same issue. 
Do you know which commit of scummvm is based libretro-scummvm for i try with the standalone i have same issue?

**Comment by spleen1981** [comment]
> Do you know which commit of scummvm is based libretro-scummvm for i try with the standalone i have same issue?

The libretro core is part of scummvm official source, so you can use the same commit to build the standalone version.

> i had scummvm_video_hw_acceleration = "disabled" and still same issue.

It is quite strange you have the same graphical issue, it's two completely different scummvm graphical managers.

**Comment by beudbeud** [comment]
I have this problem only on few games.

Maniac Mansion > Amiga, Atari ST, Macintosh, PC version (LFL)
Indiana Jones and Fate of Atlantis > PC Floppy 
Zack mcKracken > Amiga, Atari ST, FM-TOWNS, Mac, PC (LFL)

**Comment by beudbeud** [comment]
But Indiana Jones and the Last Crusade: The Graphic Adventure, The Dig works well even this use scumm backend

**Comment by beudbeud** [comment]
I build scummvm with same commit and game works well, i'm totally lost why the libretro core not work

**Comment by spleen1981** [comment]
If you are 100 percent sure no patch is applied to the core, I would say it should depend on retroarch build or video driver, or related to the crt output in some way.
The core opengl mode works almost exactly as the standalone version, so there should be no difference core side. 
Maybe you could double check if with hdmi is working or test with an older version of retroarch.
Unfortunately I'm not able to reproduce this on any platform including android, win, linux, arm and aarch64, though I don't have a rpi4 or crt to test.

**Comment by beudbeud** [comment]
I'm sure I not apply any patches, I use the make file from sources without any recalbox build system. 

i will buy micro HDMI adapter and try it with raspberry pi4

on my raspberry pi 0 2w with + Gpi case 2w I haven't this issues 

on x86 no issues

it's look only on pi5 and pi4, I will confirm about HDMI

**Comment by beudbeud** [comment]
I will get hdmi adaptor today, but i still continue to do my tests, and with Zack mcKracken i get framerate go down to to 40 fps at every new scene.  And the artefact come during translation between to scene

**Comment by beudbeud** [comment]
hi I just get pi5 and micro adapter and I have same problem than pi4 and crt
![PXL_20241130_134145103 MV](https://github.com/user-attachments/assets/1359a1e0-f5ba-4730-a704-f0f98c88a929)

**Comment by spleen1981** [comment]
Can you test the `scummvm_libretro.so` from [this bundle](https://github.com/spleen1981/retroarch-kodi-addon-CoreELEC/releases/download/v1.7.3/script.retroarch.launcher.Amlogic-no.aarch64-v1.7.3.zip)?
It's in `lib/libretro` folder.

**Comment by beudbeud** [comment]
It's work perfectly. Do you know which commit it's used and how they build it?

**Comment by spleen1981** [comment]
It's latest commit (e1a7e2931877937a04e55f45a9eec6e6f7d471ff).
It's just compiled using `platform=armv8-neon` instead of `platform=rpi4_64`, that mainly differs for using `cortex-a53` flag instead of `cortex-a72` (which I doubt will make any difference for this issue).
So if you try to compile on your system using the `armv8-neon` platform and you still have the issue, it will be confirmed that it depends on the possible patches applied, as I suspected from the beginning, or something else on your build system.

**Comment by beudbeud** [comment]
I spent all my day to build the core with many options, I find what is the problem. If I build it with gcc14 I have issue, if I build with gcc13 I haven't.

Do you know if the core already be test with gcc14?

**Comment by spleen1981** [comment]
If that is the case you may test if it is a matter of optimization, you should change manually this line https://github.com/libretro/scummvm/blob/master/backends%2Fplatform%2Flibretro%2FMakefile#L518
testing in order -O2, -O1 and -O0.
Builds are not tested systematically with various gcc versions btw (not by me anyway), as long as it builds fine with buildbot and lakka build system.

**Comment by beudbeud** [comment]
with -O2 it's work good. Thanks for your help

**Comment by spleen1981** [comment]
fixed by 7310d4e9f5d11553c6c5499911bd2f9b8ff3db3b
